Geometry/HGCalCommonData/plugins/DDHGCalHEAlgo.cc

```diff
@@ -79,163 +169,380 @@ void DDHGCalHEAlgo::initialize(const DDNumericArguments & nArgs,
-    ++copyNumber[ii];
-    int   ityp     = type[ii];
-    type[ii]       =-ityp;
-    double zi      = zMinBlock[ii];
-    zMinBlock[ii] += thickModule;
-    if (heightType[i] == 0) zz = zi;
-
-    double zlayer  = zz + 2*thickModule;
-    if((i % 6)>2) zlayer  = zz + thickModule;
-
-    double zo      = zi + thick[ii];
-    double rinF    = zi * slopeB;
-    double rinB    = zlayer * slopeB;
-
-    double routF   = (heightType[i] == 0) ? rMax(zi) : rMax(zz);
-    if((i % 6)>2) routF   = (heightType[i] == 0) ? rMax(zi-thickModule) : rMax(zz-thickModule);
-    
-    double routB   = rMax(zo);
-    std::string name = "HGCal"+names[ii]+std::to_string(copy);
-    edm::LogInfo("HGCalGeom") << "DDHGCalHEAlgo test: Layer " << i << ":" 
-			      << ii << ":" << ityp << " Front " << zi << ", " 
-			      << rinF << ", " << routF << " Back " << zo 
-			      << ", " << rinB << ", " << routB;
-    DDHGCalHEAlgo::HGCalHEPar parm = (ityp == 0) ?
-      parameterLayer(rinF, routF, rinB, routB, zi, zo) :
-      parameterLayer(ityp, rinF, routF, rinB, routB, zi, zo);
-    DDSolid solid = DDSolidFactory::trap(DDName(name, idNameSpace), 
-					 0.5*thick[ii], parm.theta,
-					 parm.phi, parm.yh1, parm.bl1, 
-					 parm.tl1, parm.alp, parm.yh2,
-					 parm.bl2, parm.tl2, parm.alp);
-
-    DDName matName(DDSplit(materials[ii]).first,
-                   DDSplit(materials[ii]).second);
-    DDMaterial matter(matName);
-    DDLogicalPart glog = DDLogicalPart(solid.ddname(), matter, solid);
-          edm::LogInfo("HGCalGeom") << "DDHGCalHEAlgo test: " 
-			      << solid.name() << " Trap made of " << matName
-			      << " of dimensions " << 0.5*thick[ii] << ", "
-			      << parm.theta/CLHEP::deg << ", " 
-			      << parm.phi/CLHEP::deg << ", " << parm.yh1 
-			      << ", " << parm.bl1 << ", " << parm.tl1 
-			      << ", " << parm.alp/CLHEP::deg << ", " 
-			      << parm.yh2 << ", " << parm.bl2 << ", " 
-      			      << parm.tl2 << ", " << parm.alp/CLHEP::deg;
-    DDTranslation r1(parm.xpos, parm.ypos, parm.zpos);
-    cpv.position(glog, module, copy, r1, rot);
-    edm::LogInfo("HGCalGeom") << "DDHGCalHEAlgo test: " << glog.name()
-			      << " number " << copy << " positioned in " 
-			      << module.name() << " at " << r1 << " with " 
-			      << rot;
-  }   // End of loop on layers
+      std::string name = "HGCal"+names_[ii]+std::to_string(copy);
+#ifdef EDM_ML_DEBUG
+      edm::LogVerbatim("HGCalGeom") << "DDHGCalHEAlgo: Layer " << ly << ":" 
+				    << ii << " Front " << zi << ", " << routF
+				    << " Back " << zo << ", " << rinB 
+				    << " superlayer thickness "
+				    << layerThick_[i];
+#endif
+      DDName matName(DDSplit(materials_[ii]).first, 
+		     DDSplit(materials_[ii]).second);
+      DDMaterial matter(matName);
+      DDLogicalPart glog;
+      if (layerSense_[ly] == 0) {
+	double alpha = CLHEP::pi/sectors_;
+	double rmax  = routF*cos(alpha) - tol;
+	std::vector<double> pgonZ, pgonRin, pgonRout;
```

Comment:
do these need to be vectors? if they only have 2 elements each, they could just be `std::array`s.